tests/integration_tests/Test/PM/Integration/_data/rest-app/module.xml

```diff
@@ -23,9 +23,8 @@
         Path="/src"
         Recurse="1"
         Directory="{$cspdir}/restdemo" 
-        MatchRoles=":{$dbrole}"
-        PasswordAuthEnabled="1"
-        UnauthenticatedEnabled="1"
+        MatchRoles=":${dbrole}:%SQL:${codeReadRole}:%Developer,%All:%Manager"
```

Comment:
Clarifying question: what is the format of MatchRoles? What does this end up configuring in the web application?
I'd expect it to be <user role>:<grant role>,<user role>:<grant role>,... but maybe I'm wrong here.

Reply:
So it's comma separated pieces, each piece is colon separated in the form of `MatchRole:TargetRole1:TargetRole2:TargetRole3`. 

- If MatchRole is nonempty, the target roles will always be added as long as `MatchRole` is detected. This configures the `Match Roles` tab in web portal
- If MatchRole is empty, the target roles will always be assigned. This configures the `Application Roles` tab in web portal.